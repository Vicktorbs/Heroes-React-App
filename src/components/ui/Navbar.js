import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link 
                className="navbar-brand" 
                to="/"
            >
                Asociaciones
            </Link>

            <div className="collapse navbar-collapse">
                <div className="navbar-nav me-auto mb-2 mb-lg-0">

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/dc"
                    >
                        DC
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/search"
                    >
                        Search
                    </NavLink>
                </div>
                <span className="navbar-text">
                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/login"
                    >
                        Logout
                    </NavLink>
                </span>
            </div>
        </nav>
    )
}