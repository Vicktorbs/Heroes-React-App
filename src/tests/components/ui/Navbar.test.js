import React from 'react';
import { mount } from "enzyme";
import { AuthContext } from "../../../auth/AuthContext";
import { Navbar } from "../../../components/ui/Navbar";
import { MemoryRouter, Router } from 'react-router-dom';
import { types } from '../../../types/types';

describe('Tests on <Navbar /> component', () => {
    
    // Mock para testear el uso del history
    const historyMock = {
        push: jest.fn(),
        replace: jest.fn(),
        location: {},
        listen: jest.fn(),
        createHref: jest.fn()
    }

    const contexValue = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'Víctor'
        }
    }
    
    const wrapper = mount(
        <AuthContext.Provider value={ contexValue } >
            <MemoryRouter>
                <Router history={ historyMock } >
                    <Navbar />
                </Router>
            </MemoryRouter>
        </AuthContext.Provider>
    );

    // Limpieza del mock
    afterEach(() => {
        jest.clearAllMocks();
    })

    test('should show itself correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Víctor');
    })

    test('should call logout and use history', () => {
        wrapper.find('button').prop('onClick')();

        expect(contexValue.dispatch).toHaveBeenCalledWith({
            type: types.logout
        });
        expect(historyMock.replace).toHaveBeenCalledWith('/login')
    })
        
})
