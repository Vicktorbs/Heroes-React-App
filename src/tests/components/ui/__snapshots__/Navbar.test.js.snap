// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <Navbar /> component should show itself correctly 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
  >
    Asociaciones
  </a>
  <div
    className="collapse navbar-collapse"
  >
    <div
      className="navbar-nav me-auto mb-2 mb-lg-0"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
    <span
      className="navbar-text text-info"
    >
      VÃ­ctor
    </span>
    <span
      className="navbar-text"
    >
      <button
        className="nav-item nav-link btn text-white"
        onClick={[Function]}
      >
        Logout
      </button>
    </span>
  </div>
</nav>
`;
